radix-engine/src/system/system_modules/costing/costing_module.rs

```diff
@@ -138,6 +158,16 @@ impl CostingModule {
                 .entry(key)
                 .or_default()
                 .add_assign(cost_units);
+
+            // Add an entry for the more detailed execution cost
+            cost_breakdown.detailed_execution_cost_breakdown.push((
+                self.current_depth,
```

Comment:
Sounds good, will do that.

Reply:
I would recommend leaving this change for a later PR since it's quite a big change and the logic around the depth I think is somewhat simple and can easily be verified.

Reply:
Looks like you did fix this in the end, thanks :).

But yeah, I think it makes sense to change in this PR so we don't introduce something and then remove it which creates unneeded churn.